src/data/real/irrational.lean

```diff
@@ -22,6 +22,20 @@ open rat real multiplicity
 /-- A real number is irrational if it is not equal to any rational number. -/
 def irrational (x : ℝ) := x ∉ set.range (coe : ℚ → ℝ)
 
+lemma irrational_iff_ne_rational (x : ℝ) : irrational x ↔ ∀ a b : ℤ, 0 < b → x - a / b ≠ 0 :=
```

Comment:
Why do you want to use `x - a / b ≠ 0` instead of `x ≠ (a / b : ℝ)` here? As for the proof, do we have something like `rat.forall {p : ℚ → Prop} : (∀ r : ℚ, p r) ↔ ∀ a b : ℤ, 0 < b → p (a / b)`? If not yet, then we should. This lemma would simplify the proof to something like `by simp [irrational, rat.forall, rat.cast_mk]` (haven't tested).

Reply:
This formulation is used in Liouville's theorem. In definition of a liouville's number, I need to have access to numerator and denominator directly. I am in process of porting liouville's theorem and transcendence of e.

Reply:
I can use `x\ne (a / b : R)`. Let me change it.

Reply:
Please see the newest commit @urkud .